pkg/quota/generic/evaluator.go

```diff
@@ -81,22 +84,39 @@ func MatchesNoScopeFunc(scope api.ResourceQuotaScope, object runtime.Object) (bo
 }
 
 // Matches returns true if the quota matches the specified item.
-func Matches(resourceQuota *api.ResourceQuota, item runtime.Object, matchFunc MatchingResourceNamesFunc, scopeFunc MatchesScopeFunc) (bool, error) {
+func Matches(resourceQuota *api.ResourceQuota, item runtime.Object,
+	matchFunc MatchingResourceNamesFunc, scopeFunc MatchesScopeFunc,
+	priorityClassScopeFunc MatchesPriorityClassScopeFunc,
```

Comment:
the signature for scopeFunc and priorityClassScope Func are equivalent

cant the pod Matches just pass a single scope func that knows how to evaluate all scope types?  in practice, pods are the only thing that use scopes now anyway.

Reply:
We can merge both into one but then will have to pass `matchPriorityClassScopes` to the combined `scopeFunc`. If this bool val is true, then pod object must match one of the priority class scopes. To avoid this complexity and reducing `if-else`, thought of keeping separate funcs. I do realize that still there is new `if` block being introduced in [`Matches`](https://github.com/kubernetes/kubernetes/pull/57963/files#diff-df7430c6b773416a0aa76f1f627d1405R107).

Please suggest if you think otherwise.